### PR TITLE
TCVP-1954 Created JJDispute mapper and tests

### PR DESCRIPTION
**Pull request by KDNeufeld**
# Description

This PR includes the following proposed change(s):

TCVP-1954
Created a JJDisputeMapper for JJDispute and JJDisputeCount.
There are several mis-matched fields that either exist in ORDS and not in main model or visa versa. For now, these have been flagged with a TODO.

Added a JJDisputeMapperTest to confirm mapping.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

mvn verify

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	@Mapping(source = "lawyerGiven3Nm", target = "lawyerGivenName3")
+	@Mapping(source = "lawFirmNm", target = "lawFirmName")
+	@Mapping(source = "interpreterLanguageCd", target = "interpreterLanguageCd")
+	@Mapping(source = "courtHearingTypeCd", target = "hearingType", qualifiedByName="mapHearingType")
+	@Mapping(source = "disputeCounts", target = "jjDisputedCounts")
+	@Mapping(source = "entDtm", target = "createdTs")
+	@Mapping(source = "entUserId", target = "createdBy")
+	@Mapping(source = "updDtm", target = "modifiedTs")
+	@Mapping(source = "updUserId", target = "modifiedBy")
+	public abstract JJDispute convert(ca.bc.gov.open.jag.tco.oracledataapi.ords.tco.api.model.JJDispute jjDispute);
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java:46]: Note: there is a warning in the code here which I left intentionally so these fields will get dealt with some day. These fields are not in ORDS and so have no map to the main application.

`Unmapped target properties: "courthouseLocation, disputantAttendanceType, enforcementOfficer, fineReductionReason, jjAssignedTo, jjDecisionDate, jjDisputeCourtAppearanceRoPs, remarks, timeToPayReason, vtcAssignedTo, vtcAssignedTs, witnessNo".`

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	@Mapping(source = "totalFineAmt", target = "totalFineAmount")
+	@Mapping(source = "commentsTxt", target = "comments")
+	@Mapping(source = "requestTimeToPayYn", target = "requestTimeToPay")
+	@Mapping(source = "requestReductionYn", target = "requestReduction")
+	@Mapping(source = "requestCourtAppearanceYn", target = "appearInCourt")
+	@Mapping(source = "entDtm", target = "createdTs")
+	@Mapping(source = "entUserId", target = "createdBy")
+	@Mapping(source = "updDtm", target = "modifiedTs")
+	@Mapping(source = "updUserId", target = "modifiedBy")
+	public abstract JJDisputedCount convert(ca.bc.gov.open.jag.tco.oracledataapi.ords.tco.api.model.JJDisputeCount jjDisputeCount);
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java:66]: Note: there is a warning in the code here which I left intentionally so these fields will get dealt with some day. These fields are not in ORDS and so have no map to the main application.

`Unmapped target properties: "description, jjDispute, jjDisputedCountRoP, ticketedFineAmount".`

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/JJDispute.java [hunk 3]
```diff
@@ -38,9 +36,7 @@
 @Table
 @Getter
 @Setter
-@Builder(toBuilder = true)
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/JJDispute.java:41]: Removed this builder as it was conflicting with Mapstruct.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/Plea.java [hunk 4]
```diff
@@ -3,9 +3,21 @@
...
-public enum Plea {
+public enum Plea implements ShortNamedEnum {
+
 	/** Unknown type (undefined). Must be index 0. */
-	UNKNOWN,
-	G,
-	N
+	UNKNOWN("U"),
+	G("G"),
+	N("N");
```
Comment (KDNeufeld) [inline src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/Plea.java:11]: These 2 enum values should have useful names, not just G or N.

**Review by karahanb — APPROVED** [review]
Fantastic work! That's a great start for switching to ORDS for JJDisputes.

File: src/backend/oracle-data-api/src/main/resources/tco-openapi-spec.yaml [hunk 5]
```diff
@@ -149,9 +149,9 @@ components:
         disputeCountId:
           type: integer
         disputeId:
-          type: string
+          type: integer
```
[the remark this answers is not in the document]
Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/resources/tco-openapi-spec.yaml:152]: I can, but all ids in the application's model are Longs not Strings. I was actually thinking of changing these from integers to Long. Ie. [JJDisputeCount.id](https://github.com/bcgov/jag-traffic-courts-online/blob/main/src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/model/JJDisputedCount.java#L46)

Everything coming from ORDS is string, but the application's ids are Long. What's the difference if the OpenAPI spec generates the fields as Long vs we converting them to Long in the mapper?

**Comment by KDNeufeld** [comment]
Agreed, Remarks will need it's own mapper. However this will need to be a separate PR as there is one sample record in ORDS and it doesn't have any remarks (the collection is empty). So it's currently impossible to know what the fields are for mapping.  This is why it wasn't included in this PR.

image

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java (hunk 1 above)
Comment (karahanb) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java:46]: That's weird. I was able to return remarks as populated for the ticketNumber EA90100004
remarks-response

**Review by lfdgov — APPROVED** [review]
After finishing your review, please update the

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java (hunk 1 above)
Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java:45]: :) It must have been just added. I see the same thing now.  I'll add mapping in the next PR.

File: src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java (hunk 2 above)
Comment (KDNeufeld) [reply src/backend/oracle-data-api/src/main/java/ca/bc/gov/open/jag/tco/oracledataapi/mapper/JJDisputeMapper.java:66]: Done.

**Pull request merged**